### PR TITLE
ft: assign a site per consumer

### PR DESCRIPTION
**Pull request by rahulreddy**
QueueProcessor will receive a site as an argument and will process entries belonging
to only that site.

### CODE REVIEW

File: extensions/replication/queueProcessor/task.js [hunk 1]
```diff
@@ -8,11 +8,19 @@ const config = require('../../../conf/Config');
...
 
+const site = process.argv[2];
+assert(site, 'QueueProcessor task must be started with a site as argument');
+
+const bootstrapList = repConfig.destination.bootstrapList
+    .filter(item => item.site === site);
+assert(bootstrapList.length === 1, 'Invalid site argument. Site must match ' +
+    'one of the replication endpoints defined');
+
+const destConfig = Object.assign({ bootstrapList }, {}, repConfig.destination);
```
Comment (philipyoo) [inline extensions/replication/queueProcessor/task.js]: Testing this out locally, I'm seeing  `destConfig.bootstrapList` being set to all items in `repConfig.destination.bootstrapList`. Swapping 1st and 3rd arguments seems to replace correctly

Comment (bennettbuchanan) [reply extensions/replication/queueProcessor/task.js]: Nice catch! Looks like we only want the filtered bootstrap list for `destConfig`.

Comment (rahulreddy) [reply extensions/replication/queueProcessor/task.js]: The perils of 1AM coding.Nice catch!

**Comment by bennettbuchanan** [comment]
@ironman-machine 
SCALITY_FEDERATION_BRANCH=FT/S3C-1145-consumers
try

**Comment by rahulreddy** [comment]
@ironman-machine
SCALITY_FEDERATION_BRANCH=FT/S3C-1145-consumers
try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_INTEGRATION_BRANCH=poc/echo-mode
try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_INTEGRATION_BRANCH=ft/S3C-1144/one-to-many
SCALTIY_S3_BRANCH=ft/S3C-1144/object-md-multiple-replica-versionIds
try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_S3_BRANCH=ft/S3C-1144/object-md-multiple-replica-versionIds
try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALTIY_S3_BRANCH=
try

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
@ironman-machine 
SCALITY_ARSENAL_BRANCH=fix/S3C-1144/enforce-two-roles-when-scality-endpoint
try

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_ARSENAL_BRANCH=fix/S3C-1144/enforce-two-roles-when-scality-endpoint
try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
end-to-end passed here: http://ci.ironmann.io/gh/scality/Integration/18683

**Pull request merged**